src/routes/products/functions/(components)/MultiFrameworkCode.svelte

```diff
@@ -0,0 +1,93 @@
+<script lang="ts">
+    import { platformMap } from '$lib/utils/references';
+    import { writable } from 'svelte/store';
+    import { getCodeHtml, type Language } from '$lib/utils/code';
+    import { copy } from '$lib/utils/copy';
+    import { Select, Tooltip } from '$lib/components';
+
+    export let selected: string = '';
+    export let data: { language: string; content: string; platform: string }[] = [];
+    export let height: number | null = null;
+
+    $: snippets = writable(new Set(data.map((d) => `${d.platform}`)));
+
+    $: content = data.find((d) => `${d.platform}` === selected)?.content ?? '';
+
+    $: ({ language } =
+        data.find((d) => `${d.platform}` === selected) ?? ({} as (typeof data)[number]));
+
+    snippets?.subscribe((n) => {
+        if (selected === '' && n.size > 0) {
+            selected = Array.from(n)[0];
+        }
+    });
+
+    enum CopyStatus {
+        Copy = 'Copy',
+        Copied = 'Copied!'
+    }
+    let copyText = CopyStatus.Copy;
+    async function handleCopy() {
+        await copy(content);
+
+        copyText = CopyStatus.Copied;
+        setTimeout(() => {
+            copyText = CopyStatus.Copy;
+        }, 1000);
+    }
```

Comment:
Perhaps not for now, but we have this code snippet (or very similar) in multiple different places. Might be nice to merge those together!

Reply:
Yeah, sounds like a plan! Probably will do it in a separate branch.